### PR TITLE
Integration with FCM

### PR DESCRIPTION
**Pull request by rafaelje**
### Changes description

- This is an integration with Firebase service

How to send a push notification manual:

- Go to Firebase console
- Cloud Messaging
- Send new message
- Fill the description message and one key, value with the topic information
- IMPORTANT: has a new module policy extracting the policies to this module the policies in this module are clean without any data manage or state is library to maintain the Android Policies and Custom Policies

Implement gplay and fdroid flavors
- gplay has Firebase to execute the policies without any service
- fdroid has mqtt service to execute policies on the agent

- fix instrumentation test with firebase for both version

@ajsb85 please review your credentials on Firebase and let me know if we need add someone to test it

screencapture-console-firebase-google-u-1-project-mdm-agent-87886-notification-compose-2018-10-08-21_19_18

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@rafaelje |50|

### References

Closes #N/A
Related #N/A
Depends on #401

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -168,6 +192,11 @@
                 <action android:name="org.flyve.mdm.agent.ACTION_INVENTORY"/>
             </intent-filter>
         </service>
+        <service android:name=".firebase.MyFirebaseMessagingService">
```
Comment (ajsb85) [inline app/src/main/AndroidManifest.xml]: Looks like there are just name of tests

**Comment by ajsb85** [comment]
image

**Comment by rafaelje** [comment]
With the official documentation is not possible to test the background process with the FCM console is required to create our own implementation to use I'll create a simple php to do it


https://firebase.google.com/docs/cloud-messaging/android/receive#sample-receive

**Comment by rafaelje** [comment]
I use this code to implement the FCM server
https://inducesmile.com/android/android-firebase-cloud-messaging-push-notification-with-server-admin-in-php-and-mysql-database/

Here explain where is the server key
https://stackoverflow.com/questions/37633188/firebase-401-unauthorized-error-fcm

**Comment by rafaelje** [comment]
This is the server code:

https://pastebin.com/QKFH9Rzj

**Comment by rafaelje** [comment]
Is added the possibility to executes policies with notifications on foreground and background

**Comment by rafaelje** [comment]
Added a new policy module with Android Policies and Custom Policies, all the policy logic are removed from the MDM core and added to this module

**Comment by rafaelje** [comment]
Hidde the connection status with FCM

9bccbcdb-de2d-42f1-b2f1-7e253d8994b5

**Pull request merged**